src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extension.tsx

```diff
@@ -70,27 +76,43 @@ export interface QueryEditorExtensionConfig {
    * A function that returns the query editor extension component. The component
    * will be displayed on top of the query editor in the search bar.
    * @param dependencies - The dependencies required for the extension.
-   * @returns The component the query editor extension.
+   * @returns The query editor extension component.
    */
   getComponent?: (dependencies: QueryEditorExtensionDependencies) => React.ReactElement | null;
   /**
    * A function that returns the query editor extension banner. The banner is a
    * component that will be displayed on top of the search bar.
    * @param dependencies - The dependencies required for the extension.
-   * @returns The component the query editor extension.
+   * @returns The query editor extension component.
    */
   getBanner?: (dependencies: QueryEditorExtensionDependencies) => React.ReactElement | null;
-
-  getSearchBarButton?: (
```

Comment:
i ended up changing this function name to `getQueryControlButtons`. Maybe long time ago this placed stuff near. search bar, but now it puts it in the `queryControls`. If you also look at the code below, this is renamed to `queryContrtols` even before this change. 

This is a breaking change, but this feature was marked as `@internal` before my change so I think we are safe to change it

Reply:
If I remember correctly, this was only used by the same file, so should be safe to update.

Reply:
dashboard-assistant use this too, I have raised a pr to update the name as well. 
https://github.com/opensearch-project/dashboards-assistant/blob/main/public/plugin.tsx#L299-L306